### ISSUE TITLE
ingester not works because of unknown memberlist_client.go 's `codec for key` error

### ISSUE THREAD

**Issue by Nebulazhang** [grafana/mimir#5981]
#### Describe the bug
The ingester service not works. And cannot query the data.

``
caller=memberlist_client.go:1235 level=error msg="failed to parse remote state: unknown codec for key" codec=usagestats.jsonCodec key=collectors/usagestats_token
``

#### Environment

- Infrastructure: [e.g., Kubernetes, bare-metal, laptop]
- Deployment tool: [e.g., helm, jsonnet]

Helm chart version: 4.3.0

**Comment by Nebulazhang** [comment]
querier error log:

```
expanding series: rpc error: code = Unimplemented desc = unknown service cortex.Ingester\\\"}\" ws: false; Accept: application/vnd.mimir.queryresponse+protobuf,application/json; X-Scope-Orgid: anonymous; uber-trace-id: 636cf497dffaea4a:19ea9c9a130334ac:241e59698766d82e:0; "
```

**Comment by Nebulazhang** [comment]
And I found the `codec=usagestats.jsonCodec key=collectors/usagestats_token` in the Loki project

**Comment by Nebulazhang** [comment]
Maybe I found the reason, I also deployed the loki in my cluster, and I don't know why the memberlist service confused the members of the loki and mimir..

**Comment by pstibrany** [comment]
> Maybe I found the reason, I also deployed the loki in my cluster, and I don't know why the memberlist service confused the members of the loki and mimir..

Yes, this is exactly the problem. You've run into the same incident as described here: https://grafana.com/blog/2022/09/28/inside-the-migration-from-consul-to-memberlist-at-grafana-labs/

We recommend setting the memberlist cluster labels to separate values for Mimir and Loki to make sure that their memberlist clusters don't join. (Or at least using different port.)

**Comment by Nebulazhang** [comment]
> We recommend setting the memberlist cluster labels to separate values for Mimir and Loki to make sure that their memberlist clusters don't join. (Or at least using different port.)

I can confirm that my loki and mimir have different services and labels, and this issue only appears occasionally,

**Comment by HujinoKun** [comment]
Hello @pstibrany and @Nebulazhang 

I confirm that I have also had this problem since switching from loki-distributed to loki, even though loki-memberlist has an additional label
